### PR TITLE
Cover invalid batch model-list lines

### LINKED ISSUE #758: [Bench/Eval][Batch] Implement model-list reader
## Goal
Implement shared model-list parsing for batch runs.

## Acceptance Criteria
- Parser handles comments, blank lines, explicit index syntax, and duplicates.
- Parser has unit coverage for invalid lines.
- Parsed model entries preserve display index and repo id.

Parent: #756

### PR DESCRIPTION
**Pull request by Zigfreidish**
## Summary
- Add targeted batch model-list parser coverage for empty display indexes and empty repo ids.
- Lock the current invalid-line usage errors so malformed `index|repo_id` entries fail before dry-run artifacts are written.
- Address review feedback by making fixtures explicit, separating comment auto-index coverage, and adding a no-trailing-newline sentinel.

## Plan or Spec
- `docs/benchmark-evaluation-contract.md`
- `docs/plans/2026-05-11-bench-eval-batch-run-foundation.md`
- Closes #758

## Commands Run
```text
git diff --check
# passed before initial push and after review fixes

xcrun swift build --product melix
# passed before initial push and after review fixes:
# Build of product 'melix' complete!

xcrun swift test --filter 'MelixCLIRunnerTests/batchModelListParserRejectsInvalidLines'
# blocked locally before the selected test ran:
# tests/MelixCLITests/DiskStreamingSmokeRunnerTests.swift:2:8: error: no such module 'Testing'

xcrun swift test --filter 'MelixCLIRunnerTests/batchModelListParser'
# blocked locally before the selected tests ran:
# tests/MelixCLITests/DiskStreamingSmokeRunnerTests.swift:2:8: error: no such module 'Testing'

.build/arm64-apple-macosx/debug/melix batch run --models .runtime/model-list-invalid-smoke/missing-index.txt --run-id invalid-index-smoke --temp-root .runtime/model-list-invalid-smoke/tmp-index --output-root .runtime/model-list-invalid-smoke/out-index --dry-run
# failed as expected before initial push and after review fixes:
# Empty model index at 2.

.build/arm64-apple-macosx/debug/melix batch run --models .runtime/model-list-invalid-smoke/missing-repo.txt --run-id invalid-repo-smoke --temp-root .runtime/model-list-invalid-smoke/tmp-repo --output-root .runtime/model-list-invalid-smoke/out-repo --dry-run
# failed as expected before initial push and after review fixes:
# Empty repo id at 3.
```

## Coverage and Metrics
- Coverage: N/A: this is a Swift test-only coverage slice, and local Swift test execution is blocked by the existing `import Testing` module resolution failure before the selected tests run.
- Metrics: N/A: this slice only adds parser regression coverage for existing validation behavior; it does not change runtime behavior or performance-sensitive code.

## Known Gaps
- The focused Swift tests could not execute locally because the current local toolchain cannot compile the test target's `import Testing` dependency. CI remains the validation gate for the added tests.
- Non-dry-run batch execution remains deferred to #755 and child execution issues.

## Evidence Checklist
- [x] Relevant plan or spec is identified.
- [x] Behavior changes are reflected in the relevant docs. (N/A: this test-only slice covers existing documented model-list behavior.)
- [x] Protocol changes include regenerated generated artifacts. (N/A: no protocol changes.)
- [x] Dependency changes include updated lockfiles. (N/A: no dependency changes.)
- [x] Relevant tests were run or attempted, with the local Swift Testing blocker recorded.
- [x] A metrics report is included, or `N/A` is stated explicitly with the reason.
- [x] Deferred work and known gaps are stated explicitly.

### CODE REVIEW

File: tests/MelixCLITests/MelixCLIRunnerTests.swift [hunk 1]
```diff
@@ -388,6 +388,26 @@ struct MelixCLIRunnerTests {
         #expect(models[0]["repo_id"] as? String == "mlx-community/Ten-4bit")
     }
 
+    @Test("batch model-list parser rejects invalid lines")
+    func batchModelListParserRejectsInvalidLines() throws {
+        #expect(throws: MelixCLIError.usage("Empty model index at 2.")) {
+            _ = try BatchRunModelListParser.parse(contents: """
+            mlx-community/Valid-4bit
+              |mlx-community/MissingIndex-4bit
```
Comment (Zigfreidish) [inline tests/MelixCLITests/MelixCLIRunnerTests.swift]: The two leading spaces before `|` here are load-bearing: Swift strips the indentation of the closing `"""` (12 spaces) from every content line, leaving `  |mlx-community/MissingIndex-4bit`. That's the intended whitespace-only index — but the correctness of the test silently depends on this precise indentation alignment. If the block is ever auto-formatted or re-indented the two extra spaces vanish and the test either stops covering the whitespace case or starts testing a pipe-prefixed repo-id instead.

Consider making the intent explicit with a raw multiline string (Swift 5.9+ `#"""..."""#` with explicit `\n`) or a `String` literal with an embedded `\n`, e.g.:

```swift
_ = try BatchRunModelListParser.parse(contents: "mlx-community/Valid-4bit\n  |mlx-community/MissingIndex-4bit\n")
```

This removes the fragile dependency on indentation column counts.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: tests/MelixCLITests/MelixCLIRunnerTests.swift [hunk 2]
```diff
@@ -388,6 +388,26 @@ struct MelixCLIRunnerTests {
...
+            """)
+        }
+        #expect(throws: MelixCLIError.usage("Empty repo id at 3.")) {
+            _ = try BatchRunModelListParser.parse(contents: """
+            mlx-community/Valid-4bit
+            # comments do not consume auto indexes
+            42|
+            """)
+        }
```
Comment (Zigfreidish) [inline tests/MelixCLITests/MelixCLIRunnerTests.swift]: This sub-test is doing two things at once: verifying that comment lines don't consume auto-indexes **and** that an empty repo-id is rejected. The mixed concerns make the failure message ambiguous — if the line-number in the error ever changes, it's unclear which assumption broke.

Consider splitting into two focused sub-tests:
1. One that checks comment lines don't consume auto-indexes (assert the successful parse produces the expected index sequence).
2. One that checks `42|` produces `"Empty repo id at N."` — using a simpler, comment-free fixture so the line-count is obvious.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: tests/MelixCLITests/MelixCLIRunnerTests.swift [hunk 3]
```diff
@@ -388,6 +388,26 @@ struct MelixCLIRunnerTests {
...
+        #expect(throws: MelixCLIError.usage("Empty model index at 1.")) {
+            _ = try BatchRunModelListParser.parse(contents: "|\n")
```
Comment (Zigfreidish) [inline tests/MelixCLITests/MelixCLIRunnerTests.swift:400]: Good minimal case. Worth also testing `"|"` (no trailing newline) to confirm the parser doesn't require a terminal newline to detect the error — especially since real files fed via CLI sometimes lack a trailing newline:

```swift
#expect(throws: MelixCLIError.usage("Empty model index at 1.")) {
    _ = try BatchRunModelListParser.parse(contents: "|")
}
```

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by Zigfreidish — COMMENTED** [review]
## Review: Cover invalid batch model-list parser lines

**Scope:** Test-only change — one new `@Test` function with three `#expect(throws:)` sub-cases covering empty-index and empty-repo-id rejection.

**What's good:**
- The three cases map cleanly to the two distinct error paths (missing index, missing repo-id).
- Using `#expect(throws:)` with the exact `MelixCLIError.usage(_:)` value pins both the error type and message, which is exactly the right level of specificity for a parser regression test.
- The `"|\n"` minimal case (line 407) is a solid sentinel: it confirms both checks are order-dependent (index before repo-id) and that a one-character input is handled.
- The PR description honestly records the local `import Testing` blocker and delegates validation to CI — that's the right call.

**Concerns (see inline comments):**

1. **Fragile multiline-string indentation** (line 396) — the two leading spaces that make `  |repo` a whitespace-only index are a side-effect of Swift's closing-`"""` indentation stripping. Auto-formatting silently removes them. Prefer an explicit single-line string with `\n` escapes.

2. **Mixed concerns in second sub-test** (lines 399–405) — the fixture conflates "comment lines don't consume auto-indexes" with "empty repo-id is rejected." Split them so each failure points to exactly one broken invariant.

3. **Missing no-trailing-newline variant** (line 407) — real CLI input files sometimes lack a terminal newline; a `"|"` (no `\n`) companion case would close that gap with one extra line.

No blocking issues — the tests cover real gaps and the production code is untouched. Addressing the indentation fragility before merge would be my priority.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: tests/MelixCLITests/MelixCLIRunnerTests.swift (hunk 1 above)
Comment (Zigfreidish) [reply tests/MelixCLITests/MelixCLIRunnerTests.swift]: Addressed in 5ccfaba0: the missing-index fixture now uses an explicit string with \n escapes, so the whitespace-only index no longer depends on multiline-string indentation stripping.

File: tests/MelixCLITests/MelixCLIRunnerTests.swift (hunk 2 above)
Comment (Zigfreidish) [reply tests/MelixCLITests/MelixCLIRunnerTests.swift]: Addressed in 5ccfaba0: the empty-repo fixture is now comment-free with an obvious line number, and comment handling moved into a separate successful-parse test that asserts indexes, repo ids, and source lines.

File: tests/MelixCLITests/MelixCLIRunnerTests.swift (hunk 3 above)
Comment (Zigfreidish) [reply tests/MelixCLITests/MelixCLIRunnerTests.swift:400]: Addressed in 5ccfaba0: added the no-trailing-newline "|" sentinel alongside the existing "|\n" case.
